src/entities/image-source.js

```diff
@@ -0,0 +1,36 @@
+//@flow
+
+const BASE_THUMBNAIL_URL_TEMPLATE = '.+entry_id/[a-zA-Z0-9_]+/';
+
+export default class ImageSource {
+  /**
+   * @member - media source id
+   * @type {string}
+   */
+  id: string;
+  /**
+   * @member - media source url
+   * @type {string}
+   */
+  url: string;
+  /**
+   * @member - media source mimetype
+   * @type {string}
+   */
+  mimetype: string;
+
+  constructor(entry: Object) {
+    this.id = entry.id;
+    this.url = ImageSource.extractBaseThumbnailUrl(entry.dataUrl);
+    this.mimetype = 'image/jpeg';
+  }
+
+  /**
+   * Convert url to template url.
+   * @param {string} url - dataUrl.
+   * @returns {string} - The template url .
+   */
+  static extractBaseThumbnailUrl(url: string): string {
```

Comment:
why is it static ?

Reply:
as long as the method is instance agnostic (and the indication is that you don’t see the `this` keyword) it should be static